src/snowflake/snowpark/mock/_functions.py

```diff
@@ -929,3 +929,17 @@ def mock_to_variant(expr: ColumnEmulator):
     res = expr.copy()
     res.sf_type = ColumnType(VariantType(), expr.sf_type.nullable)
     return res
+
+
+@patch("greatest")
+def mock_greatest(*exprs: ColumnEmulator):
+    result = reduce(lambda x, y: x.combine(y, max), exprs)
+    result.sf_type = exprs[0].sf_type
+    return result
+
+
+@patch("least")
+def mock_least(*exprs: ColumnEmulator):
```

Comment:
```
The first argument determines the return type:

If the first type is numeric, then the return type is ‘widened’ according to the numeric types in the list of all arguments.

If the first type is not numeric, then all other arguments must be convertible to the first type.

If any of the argument values is NULL, the result is NULL.
```
from https://docs.snowflake.com/en/sql-reference/functions/least#usage-notes

The first two cases since those require coercion (implicit type conversion, "123" can be implicitly evaluated to `123`) for this PR, we should leave a TODO in the docstring.

Does the current implementation support the third case?

Reply:
I've modified logic and added test cases to show that coercion works correctly.

Reply:
Sorry I had a typo in the comment above, I meant that coercion is not supported in Local Testing in general so we could ignore the first two cases in this PR and simply leave a TODO in the docstring (including a JIRA ticket)